runtime/src/impls.rs

```diff
@@ -47,26 +48,32 @@ pub struct SplitToAllValidators;
 /// The reward is split evenly and distributed to all of the current elected validators.
 /// The remainder from the division are burned.
 impl OnUnbalanced<NegativeImbalance> for SplitToAllValidators {
-	fn on_nonzero_unbalanced(amount: NegativeImbalance) {
+	fn on_nonzero_unbalanced(imbalance: NegativeImbalance) {
 		let validators = Staking::<Runtime>::current_elected();
-		if validators.len().is_zero() || amount.peek().is_zero() {
+		if validators.len().is_zero() || imbalance.peek().is_zero() {
 			return;
 		}
 		// Get a list of elected validators
-		let per_validator_reward: Balance = amount.peek() / (validators.len() as Balance);
+		let per_validator_reward: Balance = imbalance.peek() / (validators.len() as Balance);
 
 		// This tracks the total amount of reward actually handed out. Used to adjust total issurance
 		let mut total_imbalance = PositiveImbalance::zero();
-		let asset_id = amount.asset_id();
-		for v in &validators {
+		let asset_id = imbalance.asset_id();
+		for validator in &validators {
+			let dest = Staking::<Runtime>::payee(validator);
+			let reward_destination_account_id = match dest {
+				RewardDestination::Controller => Staking::<Runtime>::bonded(validator).unwrap(),
```

Comment:
Maybe we want to use
```rust
Staking::<Runtime>::bonded(validator).unwrap_or_else(|| validator);
```
to ensure that if `bonded() -> None` then we just want to use validator

Reply:
Yep done.